File: app/src/main/java/com/example/owner/skymood/fragments/MoreInfoFragment.java
package com.example.owner.skymood.fragments;

import android.content.Context;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.LinearLayout;
import android.widget.ProgressBar;
import android.widget.TextView;

import com.example.owner.skymood.R;
import com.example.owner.skymood.MainActivity;

public class MoreInfoFragment extends Fragment  implements Slidable {

    private Context context;
    private String city;
    private String code;


    private TextView day;
    private TextView date;
    private TextView temp;
    private TextView feels;
    private TextView min;
    private TextView max;
    private TextView uv;
    private TextView humidity;
    private TextView pressure;
    private TextView visibility;
    private TextView sunrise;
    private TextView sunset;
    private TextView moonAge;
    private TextView moonIlluminated;
    private TextView windSpeed;
    private TextView condition;
    private TextView moonPhase;
    private ProgressBar progress;
    private LinearLayout layout;
    private TextView textCon;

    private String dayTxt;
    private String dateTxt;
    private String tempTxt;
    private String feelsTxt;
    private String minTxt;
    private String maxTxt;
    private String uvTxt;
    private String  humidityTxt;
    private String pressureTxt;
    private String windsSpeedTxt;
    private String visibilityTxt;
    private String sunriseTxt;
    private String sunsetTxt;
    private String conditionTxt;
    private String moonPhaseTxt;
    private int moonAgeTxt;
    private int moonIlluminatedTxt;
    private static final String API_KEY = "7fc23227bbbc9a36";
    private MainActivity activity;

    public MoreInfoFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View root = inflater.inflate(R.layout.fragment_more_info, container, false);

        this.day = (TextView) root.findViewById(R.id.fragment_more_info_tv_day);
        this.date = (TextView) root.findViewById(R.id.fragment_more_info_tv_date);
        this.temp = (TextView) root.findViewById(R.id.fragment_more_info_tv_temp_value);
        this.feels = (TextView) root.findViewById(R.id.fragment_more_info_tv_real_feel_value);
        this.min = (TextView) root.findViewById(R.id.fragment_more_info_tv_min_value);
        this.max = (TextView) root.findViewById(R.id.fragment_more_info_tv_max_value);
        this.uv = (TextView) root.findViewById(R.id.fragment_more_info_tv_uv_index_value);
        this.pressure = (TextView) root.findViewById(R.id.fragment_more_info_tv_pressure_value);
        this.humidity = (TextView) root.findViewById(R.id.fragment_more_info_tv_humidity_value);
        this.visibility = (TextView) root.findViewById(R.id.fragment_more_info_tv_visibility_value);
        this.sunrise = (TextView) root.findViewById(R.id.fragment_more_info_tv_sunrise_value);
        this.sunset = (TextView) root.findViewById(R.id.fragment_more_info_tv_sunset_value);
        this.moonIlluminated = (TextView) root.findViewById(R.id.fragment_more_info_tv_moon_value);
        this.moonAge = (TextView) root.findViewById(R.id.fragment_more_info_tv_moon_age_value);
        this.condition = (TextView) root.findViewById(R.id.fragment_more_info_tv_condition_value);
        this.windSpeed = (TextView) root.findViewById(R.id.fragment_more_info_tv_winds_speed_value);
        this.moonPhase = (TextView)root.findViewById(R.id.fragment_more_info_tv_phase_of_moon_value);
        this.progress = (ProgressBar) root.findViewById(R.id.fragment_more_info_view_progress_bar);
        this.layout = (LinearLayout) root.findViewById(R.id.fragment_more_info_container);
        this.textCon = (TextView) root.findViewById(R.id.fragment_more_info_tv_condition);

        //root.findViewById(R.id.more_no_internet).setVisibility(View.GONE);
            //new GetMoreInfoTask().execute();
        activity = (MainActivity) context;
        return root;
    }

    @Override
    public void setContext(Context context) {
        this.context = context;
    }

    public void setData(){
        this.temp.setText(tempTxt + "℃");
        this.feels.setText(feelsTxt + "℃");
        this.min.setText(minTxt + "℃");
        this.max.setText(maxTxt + "℃");
        this.uv.setText(uvTxt);
        this.humidity.setText(humidityTxt);
        this.pressure.setText(pressureTxt + " hPa");
        this.visibility.setText(visibilityTxt+" km");
        this.sunrise.setText(sunriseTxt);
        this.sunset.setText(sunsetTxt);
        this.moonAge.setText(moonAgeTxt+"");
        this.moonIlluminated.setText(moonIlluminatedTxt +" % illuminated");
        this.windSpeed.setText(windsSpeedTxt + " kmh");
        this.condition.setText(conditionTxt);
        this.date.setText(dateTxt);
        this.day.setText(city);
        this.moonPhase.setText(moonPhaseTxt);
        if(this.condition.getHeight() > textCon.getHeight())
            textCon.setHeight(this.condition.getHeight());
    }

    public void setExternalInfo(String city, String code, String date, String min, String max){
        this.city = city;
        this.code = code;
        this.minTxt = min;
        this.maxTxt = max;
        this.dateTxt = date;
    }

    public void setTaskInfo(String day, String temp, String feels, String  uv, String humidity, String pressure, String windsSpeed, String  visibility, String sunrise, String  sunset, String  condition, String  moonPhase, int moonAge, int illuminate){

        this.dayTxt = day;
        this.tempTxt = temp;
        this.feelsTxt = feels;
        this.uvTxt = uv;
        this.humidityTxt = humidity;
        this.pressureTxt = pressure;
        this.windsSpeedTxt = windsSpeed;
        this.visibilityTxt = visibility;
        this.sunriseTxt = sunrise;
        this.sunsetTxt = sunset;
        this.conditionTxt = condition;
        this.moonPhaseTxt = moonPhase;
        this.moonAgeTxt = moonAge;
        this.moonIlluminatedTxt = illuminate;


    }

    public ProgressBar getProgress() {
        return progress;
    }

    public LinearLayout getLayout() {
        return layout;
    }
}

File: app/src/main/java/com/example/owner/skymood/fragments/MoreInfoFragment.kt
package com.example.owner.skymood.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.example.owner.skymood.MainActivity
import com.example.owner.skymood.databinding.FragmentMoreInfoBinding

class MoreInfoFragment : Fragment() {
    private lateinit var binding: FragmentMoreInfoBinding
    private var city: String? = null
    private var code: String? = null

    private var dayTxt: String? = null
    private var dateTxt: String? = null
    private var tempTxt: String? = null
    private var feelsTxt: String? = null
    private var minTxt: String? = null
    private var maxTxt: String? = null
    private var uvTxt: String? = null
    private var humidityTxt: String? = null
    private var pressureTxt: String? = null
    private var windsSpeedTxt: String? = null
    private var visibilityTxt: String? = null
    private var sunriseTxt: String? = null
    private var sunsetTxt: String? = null
    private var conditionTxt: String? = null
    private var moonPhaseTxt: String? = null
    private var moonAgeTxt = 0
    private var moonIlluminatedTxt = 0
    private var activity: MainActivity? = null

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentMoreInfoBinding.inflate(inflater, container, false)
        activity = context as? MainActivity
        return binding.root
    }

    fun setData() {
        binding.fragmentMoreInfoTvTempValue.text = "$tempTxt℃"
        binding.fragmentMoreInfoTvRealFeelValue.text = "$feelsTxt℃"
        binding.fragmentMoreInfoTvMinValue.text = "$minTxt℃"
        binding.fragmentMoreInfoTvMaxValue.text = "$maxTxt℃"
        binding.fragmentMoreInfoTvUvIndexValue.text = uvTxt
        binding.fragmentMoreInfoTvHumidityValue.text = humidityTxt
        binding.fragmentMoreInfoTvPressureValue.text = "$pressureTxt hPa"
        binding.fragmentMoreInfoTvVisibilityValue.text = "$visibilityTxt km"
        binding.fragmentMoreInfoTvSunriseValue.text = sunriseTxt
        binding.fragmentMoreInfoTvSunsetValue.text = sunsetTxt
        binding.fragmentMoreInfoTvMoonAgeValue.text = moonAgeTxt.toString()
        binding.fragmentMoreInfoTvMoonValue.text = "$moonIlluminatedTxt % illuminated"
        binding.fragmentMoreInfoTvWindsSpeedValue.text = "$windsSpeedTxt kmh"
        binding.fragmentMoreInfoTvConditionValue.text = conditionTxt
        binding.fragmentMoreInfoTvDate.text = dateTxt
        binding.fragmentMoreInfoTvDay.text = city
        binding.fragmentMoreInfoTvPhaseOfMoonValue.text = moonPhaseTxt
        
        if (binding.fragmentMoreInfoTvConditionValue.height > binding.fragmentMoreInfoTvCondition.height) {
            binding.fragmentMoreInfoTvCondition.height = binding.fragmentMoreInfoTvConditionValue.height
        }
    }

    fun setExternalInfo(city: String?, code: String?, date: String?, min: String?, max: String?) {
        this.city = city
        this.code = code
        this.minTxt = min
        this.maxTxt = max
        this.dateTxt = date
    }

    fun setTaskInfo(
        day: String?,
        temp: String?,
        feels: String?,
        uv: String?,
        humidity: String?,
        pressure: String?,
        windsSpeed: String?,
        visibility: String?,
        sunrise: String?,
        sunset: String?,
        condition: String?,
        moonPhase: String?,
        moonAge: Int,
        illuminate: Int
    ) {
        this.dayTxt = day
        this.tempTxt = temp
        this.feelsTxt = feels
        this.uvTxt = uv
        this.humidityTxt = humidity
        this.pressureTxt = pressure
        this.windsSpeedTxt = windsSpeed
        this.visibilityTxt = visibility
        this.sunriseTxt = sunrise
        this.sunsetTxt = sunset
        this.conditionTxt = condition
        this.moonPhaseTxt = moonPhase
        this.moonAgeTxt = moonAge
        this.moonIlluminatedTxt = illuminate
    }
}
